test/utils.py

```diff
@@ -507,6 +507,49 @@ def compare_matmat(
         assert allclose_report(o_x, m_x, **tol)
 
 
+def compare_consecutive_matmats(
+    op: PyTorchLinearOperator,
+    adjoint: bool,
+    is_vec: bool,
+    num_vecs: int = 2,
+    rtol: float = 1e-5,
+    atol: float = 1e-8,
+):
+    """Compare applying the linear operator to two identical vectors in sequence.
+
+    Args:
+        op: The operator to test.
+        adjoint: Whether to test the adjoint operator.
+        is_vec: Whether to test matrix-vector or matrix-matrix multiplication.
+        num_vecs: Number of vectors to test (ignored if ``is_vec`` is ``True``).
+            Default: ``2``.
```

Comment:
Is there a case where we want `num_vecs=1` and `is_vec=False`? If yes, why? Shouldn't it be sufficient to test things with `num_vec=1` and `is_vec=True` and then `num_vec=N` where `N>1`?